[Android SDK] Edge to edge SSRActivity

kidinov opened the pull request:
<!-- Remember about a good descriptive title. -->

part: #13396
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Adapts edge to edge to SSRActivity

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
* More -> Settings -> Support and help -> Application log
* Notice edge to edge support on devices with 34 above and below in dark and light modes

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="818" alt="image" src="https://github.com/user-attachments/assets/91ba738c-0306-4d8f-ab29-f5a23aa091b8" />
<img width="818" alt="image" src="https://github.com/user-attachments/assets/f26983cc-ad6d-4b6c-ab51-84134755fb9a" />


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

kidinov committed:
Edge to edge support fo the SSRActivity

kidinov committed:
Fixed formatting

kidinov committed:
More idiomatic way of archiving of the same

atorresveiga reviewed (commented):
Thanks for working on this @kidinov.

I tested the changes and noted that the screen does not consider navigation bars. If the device have 3-button-navigation enabled, the Screen can hide some content. I will suggest including the navigation bar insets for a better experience.

I create this patch with a few enhancements. Let me know WDYT?

<details>
  <summary>Patch</summary>

```
Subject: [PATCH] fix unit tests
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/support/SSRScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/support/SSRScreen.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/support/SSRScreen.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/support/SSRScreen.kt	(revision a571f9cdb81777baf97e8069f8470e1c368b02ed)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/support/SSRScreen.kt	(date 1738618505168)
@@ -5,6 +5,7 @@
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.WindowInsets
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.navigationBars
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.statusBars
 import androidx.compose.foundation.layout.windowInsetsPadding
@@ -82,7 +83,7 @@
                     .windowInsetsPadding(WindowInsets.statusBars)
             )
         },
-        modifier = Modifier.fillMaxSize()
+        backgroundColor = MaterialTheme.colors.surface
     ) { padding ->
         val scrollState = rememberScrollState()
 
@@ -93,11 +94,11 @@
                 isLoading = isLoading,
                 formattedSSR = formattedSSR,
                 modifier = Modifier
-                    .background(color = MaterialTheme.colors.surface)
+                    .fillMaxSize()
+                    .windowInsetsPadding(WindowInsets.navigationBars)
                     .verticalScroll(scrollState)
                     .padding(padding)
                     .padding(vertical = dimensionResource(id = R.dimen.major_100))
-                    .fillMaxSize()
                     .weight(1.0f)
             )
         }
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerScreen.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerScreen.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerScreen.kt	(revision a571f9cdb81777baf97e8069f8470e1c368b02ed)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerScreen.kt	(date 1738619101656)
@@ -8,6 +8,7 @@
 import androidx.compose.foundation.layout.Row
 import androidx.compose.foundation.layout.WindowInsets
 import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.navigationBars
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.statusBars
 import androidx.compose.foundation.layout.windowInsetsPadding
@@ -78,7 +79,9 @@
     ) { padding ->
         LogViewerEntries(
             entries,
-            modifier = Modifier.padding(padding)
+            modifier = Modifier
+                .windowInsetsPadding(WindowInsets.navigationBars)
+                .padding(padding)
         )
     }
 }

```
</details>

| Before  | After |
| ------------- | ------------- |
| <img width="300" src="https://github.com/user-attachments/assets/ed1b847b-25e8-4e2c-8508-6cc7c3743a8f" />|  <img width="300" src="https://github.com/user-attachments/assets/c0b24f46-7ce5-4250-969c-da5a37047b60" /> |

kidinov committed:
Bottom padding so the while content is visible when scrolled up

kidinov commented:
@atorresveiga 👋

My understanding of the idea behind edge-to-edge is that the content should be under navigation bars, where possible. And keep in mind that probably most use gesture navigation

https://developer.android.com/develop/ui/views/layout/edge-to-edge#system-bars-insets

What I think is necessary is to add padding to the bottom so all the text is visible when scrolled up

a7367944cf584ed2ce79de885ef01d96047062f8



https://github.com/user-attachments/assets/978f9ee9-4b09-43d5-a2ef-4e85922e1203

atorresveiga commented:
@kidinov 👋 . Thanks for working on this.

> What I think is necessary is to add padding to the bottom so all the text is visible when scrolled up

I like this solution more 👍 I think this is the way to go. 

> probably most use gesture navigation

I agree with you, but I believe that even if more merchants adopt gestures, we should still ensure a good experience for those who continue using button navigation.

I found that if the screen is in the landscape, the navigation bars can hide the toolbar navigation. Can we also consider this scenario?

<img width="500" src="https://github.com/user-attachments/assets/9f5232a6-e0fe-40d3-b063-192b517bb311" />

This PR has a screenshot for the Application Log screen. I tested it and found that we will also need to add padding to this screen, but I'm not sure if this screen is also in the scope of this PR.

<img width="300" src="https://github.com/user-attachments/assets/34cc64e6-c7c9-4233-bc62-dbc8bbf83777" />

kidinov committed:
Proper handling of the landscape mode with 3 buttons navigation for SSRActivity

kidinov committed:
Proper handling of the landscape mode with 3 buttons navigation for WooLogViewerScreen

kidinov commented:
@atorresveiga 

> I agree with you, but I believe that even if more merchants adopt gestures, we should still ensure a good experience for those who continue using button navigation.

💯

> I found that if the screen is in the landscape, the navigation bars can hide the toolbar navigation. Can we also consider this scenario?

Wow, great catch. I was under the impression that the navigation bar stays on the bottom, and it does, but only for gesture navigation, not 3-button

I fixed it here

e5b71077817b7ab2296a944a1a4ceace6bafec1b

> This PR has a screenshot for the Application Log screen. I tested it and found that we will also need to add padding to this screen, but I'm not sure if this screen is also in the scope of this PR.

I put the wrong screenshot :), but as there is the same issue as above, I fixed it too

69177b30d2c6a3ce54ce2eca9603530a0a65f072

<img width="423" alt="image" src="https://github.com/user-attachments/assets/320a4ef9-4eee-4158-8df3-8386f0569375" />
<img width="837" alt="image" src="https://github.com/user-attachments/assets/0f184ee1-aab6-407a-bc9f-6aaa4c13fd05" />


<img width="429" alt="image" src="https://github.com/user-attachments/assets/c131964a-5564-4ad3-a873-f82372e18829" />
<img width="844" alt="image" src="https://github.com/user-attachments/assets/03bea743-e27b-401d-8b3e-4eadcd8cc6a5" />

atorresveiga reviewed (approved):
Outstanding work @kidinov!

There is one minor thing missing, and it is the padding at the bottom of the Application Log screen.

<img width="300" src="https://github.com/user-attachments/assets/ad4b7348-4dcd-47ed-847f-7de9af03b787" />

The changes are very similar to the ones in the SSR screen so that I will pre-approve the PR.

kidinov committed:
Fixed bottom padding for the lazy column in WooLogViewerScreen

kidinov commented:
@atorresveiga Thanks! Good catch. Fixed here c785da9b16a5c9176c08965c9bbfd41511652146
